### DISCUSSION TITLE
Remove requirement for external links / Scan individual Serial Numbers

### DISCUSSION THREAD

**Discussion by asterisktec** [inventree/InvenTree#7675] [Q&A] [answered]
1. Is it at all possible to remove the mandatory requirement for an external link for each part. While I agree that it's a useful feature, having it mandatory on each part when using the mobile app is counter intuitive and really frustrating.
2. Would it be possible the add the ability to scan barcodes to add individual serial numbers?

I appreciate any help you can offer.

**Comment by SchrodingersGat** [comment]
@asterisktec can you expand on what you mean by mandatory external link?

**Reply by asterisktec** [reply]
![URL-Error](https://github.com/user-attachments/assets/9e1b4ca9-eaed-4c62-85af-4e27afb29fda)
I deployed an instance on Monday and it's looking great, but every time I add a stock item, it asks for an External URL every time. Not sure if I've accidentally checked something or switched something on but I'm at a loss.

**Comment by asterisktec** [comment, marked as answer]
So I rebuilt my server and once I worked out why Nginx was only bound to my IPV6 address, the issue has disappeared. Not sure if I accidentally switched something something on or off but a clean install has fixed the problem. Very strange indeed.

**Reply by matmair** [reply]
Great, please reconnect if the problem pops up again

**Comment by pelrun** [comment]
I've just hit this on 0.17.10. The backend is invariably passing the field content to the url validator, even if it's blank.

**Reply by matmair** [reply]
This is nearly a year old, please open a proper bug report

**Reply by SchrodingersGat** [reply]
Is it only for the stock item form, or for other forms which have external URL fields?

**Reply by pelrun** [reply]
I've raised an issue, and I've narrowed the problem down to the strict URL validation setting. Turning it OFF makes every field using AllowedURLValidator mandatory.
